### PR TITLE
[confcom] updating diff mode and adding cs2

### PR DESCRIPTION
**Pull request by SethHollandsworth**
Diff mode was not previously robust enough to handle multiple of the same image with different commands, env vars, etc. This change allows for that to be possible and clearer by using the container name as a key instead of the image name. 

Adding configmap sidecar for additional feature in ACI

Fixing bug for `seccompProfile` field inside of `securityContext`

Also fixing a bug where you could not use regex for arm templates by leaving out a default value in the parameters.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az confcom

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
confcom

File: src/confcom/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 Release History
 ===============
+0.3.5
```
Comment (AllyW) [inline src/confcom/HISTORY.rst:5]: please update version as bot recommended.

Comment (SethHollandsworth) [reply src/confcom/HISTORY.rst:5]: Hi, we are waiting for a particular feature to be ready and merged to make a 1.0.0 version. Thanks!

**Comment by SethHollandsworth** [comment]
update: the feature we are waiting on should be done and published by the end of this month. So 0.3.5 will probably be the last version before 1.0.0

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**